### ISSUE TITLE
Ability to upload image to be shared

### ISSUE THREAD

**Issue by gi-yt** [jtheoof/swappy#148]
The ability to upload an image so it can be shared with others (like how flameshot does it) would be greatly appreciated.
You can use 0x0.st / imgur for this.

**Comment by ashdevelops** [comment]
Would like this too.

**Comment by Exoutia** [comment]
I also would love to have that option.

**Comment by tkna91** [comment]
screenshot-upload.sh
```
#!/bin/bash
set -x

file=~/tmp/$(date "+%Y%m%d-%H%M%S")_screenshot.png
grim -g "$(slurp)" - | swappy -f - -o - | pngquant - -o "$file" &&
yad --title="Upload OK?" --text="Upload OK?" --image="dialog-question" --default-cancel &&
curl -F"file=@$file" https://0x0.st | wl-copy -n &&
notify-send --expire-time=3000 "Uploaded and Copied to clipboard." "$(wl-paste)"
mpv --image-display-duration=3 "$(wl-paste)"
```

**Comment by Exoutia** [comment]
video is corrupt but the scrips should work i guess. let me test it out.

**Comment by Exoutia** [comment]
![image](https://github.com/jtheoof/swappy/assets/94464106/1ff46b3a-3609-46a8-8f22-6f93feba0787)
na didn't work out for me can you help me with it @tkna91 bro

**Comment by tkna91** [comment]
@Exoutia This is not an issue here, so please write to my project issue if you can't solve it after this.
https://github.com/tkna91/shellscripts/issues
If you don't have the command, please install it with pacman or paru, etc.

**Comment by Exoutia** [comment]
> @Exoutia This is not an issue here, so please write to my project issue if you can't solve it after this.
> https://github.com/tkna91/shellscripts/issues
> If you don't have the command, please install it with pacman or paru, etc.
> 

Sorry for that I am going to do what you suggested. 😅

**Comment by rubiin** [comment]
This would be very useful in my daily workflow

**Comment by jtheoof** [comment]
Yes indeed. The goal of swappy is to be lightweight and serve one purpose. Unless there is a Linux standard for sharing (which I doubt) it's unlikely to happen in swappy. Will close this request. But thanks though.
